sparta/sparta/resources/Buffer.hpp

```diff
@@ -895,6 +877,85 @@ namespace sparta
             validator_->resizeIteratorValidator(resize_delta_, data_pool_);
         }
 
+        template<typename U>
+        iterator push_backImpl_(U&& dat)
+        {
+
+            // Check to see if the vectors need to be resized and relinked.
+            if(SPARTA_EXPECT_FALSE(is_infinite_mode_)) {
+                resizeInternalContainers_();
+            }
+            sparta_assert(numFree(), "Buffer exhausted");
+            sparta_assert(free_position_ != nullptr);
+            free_position_->data = std::forward<U>(dat);
+            free_position_->physical_idx = num_valid_;
+
+            // Create the entry to be returned.
+            iterator entry(this, free_position_);
+
+            // Do the append now.  We can do this with different logic
+            // that does not require a process.
+            buffer_map_[num_valid_] = free_position_;
+
+            // Store the index in the data_pool_ to which current
+            // free_position_ points to. We need to relink all these
+            // pointers once the data_pool_ resizes.
+            address_map_[num_valid_] =
+                static_cast<uint32_t>(free_position_ - &data_pool_[0]);
+            //Mark this data pointer as valid
+            validator_->attachDataPointer(free_position_);
+            ++num_valid_;
+            free_position_ = free_position_->next_free;
+            updateUtilizationCounters_();
+
+            return entry;
+        }
+
+        template<typename U>
+        iterator insertImpl_(uint32_t idx, U&& dat)
```

Comment:
I wonder if this function can call the above one before pushing everything down...